cmd/pebble/cmd_enter.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/jessevdk/go-flags"
+)
+
+const shortEnterHelp = "Run pebble command in new pebble environment"
+const longEnterHelp = `
+The enter command behaves just like run when run without positional arguments,
+with the exception that the --hold flag is not supported. That is, it runs as
+service manager. When run with positional arguments, the service manager is
+started in background without starting autostart services (i.e. like run with
+the --hold flag) and positional arguments are executed as pebble command. Only
+subset of pebble commands is supported, namely exec, help, ls, plan, services,
+start, stop and version. Autostart services can be started with the --run flag.
+stop command must be run with the --run flag and ls, plan and services commands
+cannot be run with the --run flag. Normally, the service manager is shut down
+when the command finishes. With start and stop commands the service manager
+continues running. The enter command is intended to be used as entrypoint in
+container images.
+`
+
+type enterFlags int
+
+const (
+	enterFlagSilcenceLogging enterFlags = 1 << iota
+	enterFlagNoServiceManager
+	enterFlagKeepServiceManager
+	enterFlagRequireServiceAutostart
+	enterFlagProhibitServiceAutostart
+)
+
+type haveEnterFlags interface {
+	enterFlags() enterFlags
```

Comment:
I've reverted the changes to individual comments and moved the logic into single type switch. Unfortunately, it's a type switch, and not a data structure, because of the way how go-flags `flags.Commander` works. This is an interface with `Execute(args []string) error` as the only method. By default, when the parser runs, this method is executed for the last command given. The only way to intercept this is to set `Parser.CommandHandler` to a function with the following signature:
```
func(command Commander args []string) error
````
IMHO, the signature should've been:
```
func(command Command, commander Commander args []string) error
```
This would give me a way to index some data structure via `command.Name`.